scaling_performance/optimizing_storage.adoc

```diff
@@ -12,6 +12,86 @@
+{product-title} cluster application and useful attributes.
+[format="csv",width="70%",cols="8"]
+[options="header]
+|=======================
+Storage Type,ROX,RWX,Registry,Scaled Registry,Metrics,Logging,Apps
+Block,Yes^(1)^,No,Configurable,Not Configurable,Recommended,Recommended,Recommended
+File,Yes^(1)^,Yes,Configurable,Configurable,Configurable,Configurable,Recommended
+Object,Yes,Yes,Recommended,Recommended,Not Configurable,Not Configurable,Not Configurable^(2)^
+|=======================
+NOTE: A Scaled Registry is a {product-title} Registry where 3 or more pod replicas are running.
+
+NOTE: ^(1)^ This does not apply to Physical Disk, VM Physical Disk, VMDK, Loopback over NFS, AWS EBS, and Azure Disk.
+
+NOTE: ^(2)^ Object Storage is not consumed via {product-title}'s PVs/PVCs.  Apps must integrate with the Object Storage REST API.
+
+==== Specific Application & Storage Recommendations
+===== Registry
+In a non-scaled/HA {product-title} Registry cluster deployment, the preferred Storage Backend is Object Storage
+followed by Block Storage.  The Storage Backend does not need to support ReadWriteMany Access Mode (RWX).  The Storage Backend *must
+ensure Read-After-Write-Consistency*.  All Network Attached File Storage Backends (excluding CNS/CRS GlusterFS) are expressly *not recommended* for {product-title} Registry cluster deployment with production workloads.  While hostPath volumes are configurable for a non-scaled/HA {product-title} Registry cluster deployment, they are not recommended.
+
+WARNING: Corruption may occur when using NFS to back {production-title} Registry at production workloads.
+
+===== Scaled Registry
+In a scaled/HA {product-title} Registry cluster deployment, the preferred Storage Backend is Ojbect Storage.
+The Storage Backend *must support ReadWriteMany* Access Mode (RWX) and *must ensure Read-After-Write-Consistency*.  Block Storage is
+expressly *not recommended* for a scaled/HA {product-title} Registry cluster deployment with production workloads.  All Network Attached File
+Storage Backends (excluding CNS/CRS GlusterFS) are expressly *not recommended* for {product-title} Registry cluster deployment with production workloads.
+
+WARNING: Corruption may occur when using NFS to back {production-title} scaled/HA Registry at production workloads.
+
+===== Metrics
+In {product-title} Hosted Metrics cluster deployment, the preferred Storage Backend is Block Storage.  All Network Attached File
+Storage Backends (excluding CNS/CRS GlusterFS) are expressly *not recommended* for Hosted Metrics cluster deployment with production workloads.
+
+WARNING: Corruption may occur when using NFS to back Hosted Metrics at production workloads.
+
+===== Logging
+In {product-title} Hosted Logging cluster deployment, the preferred Storage Backend is Block Storage.  All Network Attached File
+Storage Backends (excluding CNS/CRS GlusterFS) are expressly *not recommended* for Hosted Logging cluster deployment with production workloads.
```

Comment:
AFAIK NFS is the dangerous because lucene relies on functionality NFS doesn't have, however I don't think gluster is a good idea either nor recommended.

Gluster is good at big sequential reads/writes, elastic requires low latency small reads/writes.

Reply:
I agree that we need to prove out GlusterFS at VERY LARGE WORKLOADS but we're currently communicating that we're supporting it for Production Workloads.  If we find a ceiling somewhere, I'll be the first one to volunteer a PR.

Reply:
Gluster support for metrics and logging doesn't use the normal file storage, it uses new support for block storage via iscsi.

Reply:
I've now noted this.